src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmParsingUtilities.cs

```diff
@@ -3,43 +3,52 @@
 using System;
 using System.IO;
 using System.Linq;
-using System.Threading.Tasks;
 using global::NuGet.Versioning;
 using Microsoft.ComponentDetection.Contracts;
 using Microsoft.ComponentDetection.Contracts.TypedComponent;
 using YamlDotNet.Serialization;
 
 public static class PnpmParsingUtilities
 {
-    public static async Task<PnpmYaml> DeserializePnpmYamlFileAsync(IComponentStream file)
+    public static string DeserializePnpmYamlFileVersion(string fileContent)
     {
-        var text = await new StreamReader(file.Stream).ReadToEndAsync();
         var deserializer = new DeserializerBuilder()
             .IgnoreUnmatchedProperties()
             .Build();
-        return deserializer.Deserialize<PnpmYaml>(new StringReader(text));
+        return deserializer.Deserialize<PnpmYamlVersion>(new StringReader(fileContent)).lockfileVersion;
     }
 
-    public static DetectedComponent CreateDetectedComponentFromPnpmPath(string pnpmPackagePath)
+    public static PnpmYamlV5 DeserializePnpmYamlV5File(string fileContent)
     {
-        var (parentName, parentVersion) = ExtractNameAndVersionFromPnpmPackagePath(pnpmPackagePath);
-        return new DetectedComponent(new NpmComponent(parentName, parentVersion));
+        var deserializer = new DeserializerBuilder()
+            .IgnoreUnmatchedProperties()
+            .Build();
+        return deserializer.Deserialize<PnpmYamlV5>(new StringReader(fileContent));
     }
 
-    public static bool IsPnpmPackageDevDependency(Package pnpmPackage)
+    public static PnpmYamlV6 DeserializePnpmYamlV6File(string fileContent)
     {
-        if (pnpmPackage == null)
-        {
-            throw new ArgumentNullException(nameof(pnpmPackage));
-        }
+        var deserializer = new DeserializerBuilder()
+            .IgnoreUnmatchedProperties()
+            .Build();
+        return deserializer.Deserialize<PnpmYamlV6>(new StringReader(fileContent));
+    }
 
-        return string.Equals(bool.TrueString, pnpmPackage.dev, StringComparison.InvariantCultureIgnoreCase);
+    /// <summary>
```

Comment:
Nothing in here had doc comments before, but it seems like docs would be good so I added some. he projects configuration seems to like this format so I used it, but I'm not sure what the policy on documentation is.

Reply:
We appreciate any new docs being added, and yes this is the format we use 😄